src/dope-node/communication/communication_handler.go

```diff
@@ -0,0 +1,126 @@
+package communication
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/gorilla/websocket"
+)
+
+var knownAddresses = []string{}
+
+func RegisterEndpoints(boostrapServerAddress string, isBoostrapServer bool, port int) {
+	http.HandleFunc("/ws", wsHandler)
+	log.Println("Running server")
+	if isBoostrapServer {
+		http.HandleFunc("/addresses", httpHandler)
+	} else {
+		// This sends 'hello' message to all known nodes
+		addresses := fetchNodeAddresses(boostrapServerAddress, "127.0.0.1:"+strconv.Itoa(port))
+		for _, addr := range addresses {
+			u := url.URL{Scheme: "ws", Host: addr, Path: "/ws"}
+			conn, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+			if err != nil {
+				log.Printf("Failed to connect to %s: %v", addr, err)
+				continue
+			}
+			defer conn.Close()
+
+			messageToSend, _ := json.Marshal(Message{Content: "hello"})
+			sendMessage(*conn, messageToSend)
+		}
+	}
+
+	http.ListenAndServe(":"+strconv.Itoa(port), nil)
+}
+
+func wsHandler(w http.ResponseWriter, r *http.Request) {
+	connection := getWebsocketConnection(w, r)
+
+	for {
+		_, msg, err := connection.ReadMessage()
+		if err != nil {
+			if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure) {
+				log.Println("WebSocket closed unexpectedly:", err)
+			} else {
+				log.Println("Error reading message:", err)
+			}
+			break
+		}
+
+		log.Println("Handling WebSocket message from:", r.Host)
+		var receivedMessage Message
+		err = json.Unmarshal(msg, &receivedMessage)
+		if err != nil {
+			log.Println("Error unmarshalling message:", err)
+			continue
+		}
+		fmt.Println("Received message content:", receivedMessage.Content)
+	}
+}
+
+func httpHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I hate nitpicking function and variable naming, but `httpHandler` is pretty general in this case